### PR TITLE
Allow for optional domain directory write permissions

### PR DESCRIPTION
**Pull request by ev0net**
For security reasons, my applications require writeable storage at /home/user/web/domain.tld/. I've been hacking hestia on install/upgrade to achieve this. I'd like to see it integrated. 

/home/user/web/domain.tld/ writability will only occur when the user adds optional: `DOMAINDIR_WRITABLE='yes'` to `/usr/local/hestia/conf/hestia.conf`. Additionally, the user will have to modify their open_basedir in their php templates.

The way I've changed the hestia code does not impact default functionality of non-writable /home/user/web/domain.tld/ at all and does not require a new hestia.conf key=value to maintain this default functionality.

Allowing a writable /home/user/web/domain.tld/ is common for panels and providers. While I understand hestia allows for a sub directory under public_html and setting that directory to docroot, this is unnecessary hassle with negligible security benefits.

### CODE REVIEW

**Comment by jaapmarcus** [comment]
Why not use /home/user/web/domain.tld/private ?

**Comment by jaapmarcus** [comment]
Just checked the commit History ... And it was since the start already like this..

I would personally have an option added to hestia.conf at least

See Example:
https://github.com/hestiacp/hestiacp/blob/8951bde7c095c6778c2b03d1cd48f2cb092a3ac1/func/syshealth.sh#L537-L540

Just add it to the list ...

**Comment by ev0net** [comment]
> While I understand hestia allows for a sub directory under public_html and setting that directory to docroot, this is unnecessary hassle with negligible security benefits.

> Why not use /home/user/web/domain.tld/private ?

I choose not to speak to that in the sentence above. I understand that is an option also.

I would like to maintain more standard compatible infrastructure and write to the domain dir and not need to maintain different application directories differences for different systems. Again Allowing a writable /home/user/web/domain.tld/ is common for panels and providers. I haven't run across this restriction before.

> Just checked the commit History ... And it was since the start already like this..

The history of vesta/hestia? The domaindir used to writable?

> Just add it to the list ...

Will do. Not adding it was my attempt to have a minimal impact.

**Comment by jaapmarcus** [comment]
It was made like this 11 years ago. I have zero ideas why with the only exception to prevent users renaming public_html and other folders

**Comment by ev0net** [comment]
> prevent users renaming public_html and other folders

Yes. I remember finding that strange when I first was exploring hestia... But given the way subdomains work in hestia, keeping public_html non-changable works fine for me.

**Comment by ev0net** [comment]
@jaapmarcus Do you think you will approve this? If not, I need to include this as a hack in my workflow. Thanks!

**Comment by jaapmarcus** [comment]
Please allow a few hours/days before we merge things?

**Comment by ev0net** [comment]
Great! I just wanted to know if it was going to be included at all. Thanks!

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**
